setuptools/command/bdist_wheel.py

```diff
@@ -280,6 +281,15 @@ def finalize_options(self):
         ):
             raise ValueError(f"py-limited-api must match '{PY_LIMITED_API_PATTERN}'")
 
+        if "t" in sys.abiflags:
+            SetuptoolsWarning.emit(
+                summary=f"Ignoring `py_limited_api={self.py_limited_api!r}`.",
+                details="`Py_LIMITED_API` is currently incompatible with "
+                f"`Py_GIL_DISABLED` ({sys.abiflags=!r}).",
+                see_url="https://github.com/python/cpython/issues/111506",
+            )
+            self.py_limited_api = False
```

Comment:
I don't know how to deal with the following problem however:

The extensions themselves define `py_limited_api` like in https://github.com/Legrandin/pycryptodome/blob/39b5a627dab9124da0fb9d15a2f3a19434d96565/setup.py#L286C9-L286C28.

Not sure how that will interact with this.

Maybe the best is to just raise an exception and require the developer to be mindful about the compatibility with `py_limited_api` (and change their `setup.py` accordingly to react to the current version of Python being used)?

Reply:
Does that really break, or is a "py_limeted_api" extension, compiled for the nogil intepreter, just not cross-compatible?

Reply:
This is a bit out of my comfort zone for the time being, but my understanding of the discussion on https://github.com/python/cpython/issues/111506, https://discuss.python.org/t/python-abis-and-pep-703/34018  is that there is potential for break things.

The following hints in https://peps.python.org/pep-0703/#backwards-compatibility seem to suggest to me that some parts of the limited API may currently require (or at least assume) the GIL to work properly (therefore extensions using those parts of the limited API are potentially prone to runtime errors).

> - C-API extensions that rely on the GIL to protect global state or object state in C code will need additional explicit locking to remain thread-safe when run without the GIL.
> - C-API extensions that use borrowed references in ways that are not safe without the GIL will need to use the equivalent new APIs that return non-borrowed references. Note that only some uses of borrowed references are a concern; only references to objects that might be freed by other threads pose an issue.
> - Custom memory allocators (PyMem_SetAllocator) are required to delegate the actual allocation to the previously set allocator. For example, the Python debug allocator and tracing allocators will continue to work because they delegate the allocation to the underlying allocator. On the other hand, wholesale replacing of the allocator (e.g., with jemalloc or tcmalloc) will not work correctly.
> - Python objects must be allocated through the standard APIs, such as PyType_GenericNew or PyObject_Malloc. Non-Python objects must not be allocated through those APIs. For example, it is currently acceptable to allocate buffers(non-Python objects) through PyObject_Malloc; that will no longer be allowed and buffers should instead be allocated through PyMem_Malloc, PyMem_RawMalloc, or malloc.